core/ocb2progressive.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * OCB2.0 implementation slightly modified by Yifan Gu
+ * to support progressive encryption
+ * @author Yifan Gu
+ */
+
+/** @fileOverview OCB 2.0 implementation
+ *
+ * @author Emily Stark
+ * @author Mike Hamburg
+ * @author Dan Boneh
+ */
+
+/** @namespace
+ * Phil Rogaway's Offset CodeBook mode, version 2.0.
+ * May be covered by US and international patents.
+ *
+ * @author Emily Stark
+ * @author Mike Hamburg
+ * @author Dan Boneh
+ */
+
+sjcl.mode.ocb2progressive = {
+  createEncryptor: function(prp, iv, adata, tlen, premac) {
+    if (sjcl.bitArray.bitLength(iv) !== 128) {
+      throw new sjcl.exception.invalid("ocb iv must be 128 bits");
+    }
+    var i,
+        times2 = sjcl.mode.ocb2._times2,
+        w = sjcl.bitArray,
+        xor = w._xor4,
+        checksum = [0,0,0,0],
+        delta = times2(prp.encrypt(iv)),
+        bi, bl,
+        datacache = [],
+        pad;
+
+    adata = adata || [];
+    tlen  = tlen || 64;
+
+    return {
+      process: function(data){
+        var datalen = sjcl.bitArray.bitLength(data);
+        if (datalen == 0){ // empty input natrually gives empty output
+          return [];
+        }
+        var output = [];
+        datacache = datacache.concat(data);
+        for (i=0; i+4 < datacache.length; i+=4) {
+          /* Encrypt a non-final block */
+          bi = datacache.slice(i,i+4);
+          checksum = xor(checksum, bi);
+          output = output.concat(xor(delta,prp.encrypt(xor(delta, bi))));
```

Comment:
This is also the case for original ocb2 mode. I looked at the code and believe it is not exactly easy combine encryptor and decryptor. 
Also, some of the lines in 29-36 are simple aliasing and will be properly handled by minifiers, I believe.

ocb2 and ocb2progressive can be combined though, but it will involve wrapping ocb2 around ocb2progressive and I am wondering if this is OK.